### ISSUE TITLE
Kafka :Missing required configuration "bootstrap.servers"

### ISSUE THREAD

**Issue by joydeb28** [FusionAuth/fusionauth-issues#649]
## Kafka Integration: configuring the bootstrap server with hostname "kafka" seems not working

Description
In my docker test network, the Kafka container is running side-by-side with fusionauth (1.16.1)

Steps to reproduce

1. Log into FusionAuth
2. Go to Integrations -> Kafka
3. Configure the producer configuration section with bootstrap.servers=kafka:29092
4. Press "Send test event"

You should get the below error.
org.apache.kafka.common.errors.TimeoutException: Timeout expired while fetching topic metadata
Timeout expired while fetching topic metadata

### Platform
 - OS: Linux
 - Browser + version : chrome, FusionAuth:1.16.1

### Error Logs
fusionauth_1  | org.apache.kafka.common.config.ConfigException: Missing required configuration "bootstrap.servers" which has no default value.
fusionauth_1  | 	at org.apache.kafka.common.config.ConfigDef.parseValue(ConfigDef.java:463)
fusionauth_1  | 	at org.apache.kafka.common.config.ConfigDef.parse(ConfigDef.java:453)
fusionauth_1  | 	at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:62)
fusionauth_1  | 	at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:75)
fusionauth_1  | 	at org.apache.kafka.clients.producer.ProducerConfig.<init>(ProducerConfig.java:359)
fusionauth_1  | 	at org.apache.kafka.clients.producer.KafkaProducer.<init>(KafkaProducer.java:300)
fusionauth_1  | 	at io.fusionauth.api.service.messaging.KafkaProducerBuilder.build(KafkaProducerBuilder.java:30)
fusionauth_1  | 	at io.fusionauth.api.service.messaging.KafkaProducerProvider.onCacheLoad(KafkaProducerProvider.java:43)
fusionauth_1  | 	at com.inversoft.cache.BaseCacheLoader.load(BaseCacheLoader.java:24)
fusionauth_1  | 	at io.fusionauth.api.service.cache.IntegrationConfigurationCacheLoader.run(IntegrationConfigurationCacheLoader.java:36)
fusionauth_1  | 	at com.inversoft.scheduler.InjectableRunnable.run(InjectableRunnable.java:37)
fusionauth_1  | 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
fusionauth_1  | 	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
fusionauth_1  | 	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
fusionauth_1  | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
fusionauth_1  | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
fusionauth_1  | 	at java.base/java.lang.Thread.run(Thread.java:832)

**Comment by mooreds** [comment]
Can you share your docker compose file (or however you are assigning network names), please?

**Comment by joydeb28** [comment]
@mooreds 
Below I pasted my compose fille content.
`version: '3'

services:
  zookeeper:
    image: wurstmeister/zookeeper
    ports:
      - "2181:2181"
    networks:
      - db
  kafka:
    image: wurstmeister/kafka
    ports:
      - "9092:9092"
    environment:
      KAFKA_ADVERTISED_HOST_NAME: kafka
      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
    networks:
      db:
        aliases:
          - kafka
   
      
  db:
    image: postgres:9.6
    environment:
      PGDATA: /var/lib/postgresql/data/pgdata
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
# Un-comment to access the db service directly
#    ports:
#      - 5432:5432
    networks:
      - db
    restart: unless-stopped
    volumes:
      - db_data:/var/lib/postgresql/data

  fusionauth:
    image: fusionauth/fusionauth-app:latest
    depends_on:
      - db
      - kafka
      - zookeeper
    environment:
      DATABASE_URL: jdbc:postgresql://db:5432/fusionauth
      DATABASE_ROOT_USER: ${POSTGRES_USER}
      DATABASE_ROOT_PASSWORD: ${POSTGRES_PASSWORD}
      DATABASE_USER: ${DATABASE_USER}
      DATABASE_PASSWORD: ${DATABASE_PASSWORD}
      FUSIONAUTH_MEMORY: ${FUSIONAUTH_MEMORY}
      FUSIONAUTH_SEARCH_ENGINE_TYPE: database
      FUSIONAUTH_URL: http://fusionauth:9011
      FUSIONAUTH_RUNTIME_MODE: development
    networks:
     - db
    restart: unless-stopped
    ports:
      - 9011:9011
    volumes:
      - fa_config:/usr/local/fusionauth/config

networks:
  db:
    driver: bridge

volumes:
  db_data:
  fa_config:`

**Comment by robotdan** [comment]
> Configure the producer configuration section with bootstrap.servers=kafka:29092


In your config, you have it configured port `9092` is this a typo, or are you perhaps using the wrong port in the FusionAuth Kafka configuration? 

Assuming your config should be 

```
bootstrap.servers=kafka:9092
```

**Comment by joydeb28** [comment]
Yes It was a typo.
In config I am giving.
`bootstrap.servers=kafka:9092
`

**Comment by mooreds** [comment]
Heya, can you connect to the kafka service running in docker using a local kafka client? Because I can't.

* I downloaded kafka 2.12.
* I copied and pasted your docker-compse.yml file to a local directory
* I start it up using this command: `POSTGRES_USER=postgres POSTGRES_PASSWORD=pass DATABASE_USER=fa DATABASE_PASSWORD=pass docker-compose up`
* In another terminal, I try to list my topics

`kafka/bin/kafka-topics.sh --list --bootstrap-server localhost:9092` 

Gives me 
```
[2020-05-29 15:07:17,059] WARN [AdminClient clientId=adminclient-1] Error connecting to node kafka:9092 (id: 1003 rack: null) (org.apache.kafka.clients.NetworkClient)
java.net.UnknownHostException: kafka
        at java.net.InetAddress.getAllByName0(InetAddress.java:1280)
        at java.net.InetAddress.getAllByName(InetAddress.java:1192)
        at java.net.InetAddress.getAllByName(InetAddress.java:1126)
        at org.apache.kafka.clients.ClientUtils.resolve(ClientUtils.java:110)
        at org.apache.kafka.clients.ClusterConnectionStates$NodeConnectionState.currentAddress(ClusterConnectionStates.java:403)
        at org.apache.kafka.clients.ClusterConnectionStates$NodeConnectionState.access$200(ClusterConnectionStates.java:363)
        at org.apache.kafka.clients.ClusterConnectionStates.currentAddress(ClusterConnectionStates.java:151)
        at org.apache.kafka.clients.NetworkClient.initiateConnect(NetworkClient.java:955)
        at org.apache.kafka.clients.NetworkClient.ready(NetworkClient.java:293)
        at org.apache.kafka.clients.admin.KafkaAdminClient$AdminClientRunnable.sendEligibleCalls(KafkaAdminClient.java:1018)
        at org.apache.kafka.clients.admin.KafkaAdminClient$AdminClientRunnable.processRequests(KafkaAdminClient.java:1260)
        at org.apache.kafka.clients.admin.KafkaAdminClient$AdminClientRunnable.run(KafkaAdminClient.java:1203)
        at java.lang.Thread.run(Thread.java:748)
```

and  `./kafka-topics.sh --list --bootstrap-server kafka:9092`

gives me

```
[2020-05-29 15:10:20,008] WARN Couldn't resolve server kafka:9092 from bootstrap.servers as DNS resolution failed for kafka (org.apache.kafka.clients.ClientUtils)
Exception in thread "main" org.apache.kafka.common.KafkaException: Failed to create new KafkaAdminClient
        at org.apache.kafka.clients.admin.KafkaAdminClient.createInternal(KafkaAdminClient.java:479)
        at org.apache.kafka.clients.admin.Admin.create(Admin.java:61)
        at kafka.admin.TopicCommand$AdminClientTopicService$.createAdminClient(TopicCommand.scala:212)
        at kafka.admin.TopicCommand$AdminClientTopicService$.apply(TopicCommand.scala:216)
        at kafka.admin.TopicCommand$.main(TopicCommand.scala:57)
        at kafka.admin.TopicCommand.main(TopicCommand.scala)
Caused by: org.apache.kafka.common.config.ConfigException: No resolvable bootstrap urls given in bootstrap.servers
        at org.apache.kafka.clients.ClientUtils.parseAndValidateAddresses(ClientUtils.java:89)
        at org.apache.kafka.clients.ClientUtils.parseAndValidateAddresses(ClientUtils.java:48)
        at org.apache.kafka.clients.admin.KafkaAdminClient.createInternal(KafkaAdminClient.java:439)
        ... 5 more
```

So I'm not sure my network configuration is working to connect kafka and zookeeper, ignoring FusionAuth. I also tried to run bash on the docker containers and ran into similar networking issues.

Can you verify you can list and create topics in the kafka docker instance without FusionAuth involved at all?

**Comment by joydeb28** [comment]
I tried with the below kafka configuration.
```
services:
  zookeeper:
    image: wurstmeister/zookeeper
    ports:
      - "2181:2181"
    networks:
      - db
  kafka:
    image: wurstmeister/kafka
    ports:
      - "9092:9092"
    environment:
      KAFKA_ADVERTISED_HOST_NAME: localhost
      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
    networks:
      - db
```
Kafka is working on my local setup.
But when I am trying to plug in the same with fusionauth docker-compose I got errors.
Do you see any configuration issues from my side? Or I miss something??

**Comment by mooreds** [comment]
Interesting. I'll take a look.

Here's a previous kafka issue, but it appears to have been resolved: https://github.com/FusionAuth/fusionauth-issues/issues/318

**Comment by mooreds** [comment]
Okay. When I run `POSTGRES_USER=postgres POSTGRES_PASSWORD=pass DATABASE_USER=fa DATABASE_PASSWORD=pass docker-compose up`

with this docker file:
```
vagrant@ubuntu-xenial:~/fusionauth/dockerkafka$ cat docker-compose.yml
version: '3'

services:
  zookeeper:
    image: wurstmeister/zookeeper
    ports:
      - "2181:2181"
    networks:
      - mynetwork
  kafka:
    image: wurstmeister/kafka
    ports:
      - "9092:9092"
    environment:
      KAFKA_ADVERTISED_HOST_NAME: kafka
      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
    networks:
      mynetwork:
        aliases:
          - kafka

  db:
    image: postgres:9.6
    environment:
      PGDATA: /var/lib/postgresql/data/pgdata
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
    networks:
      - mynetwork
    restart: unless-stopped
    volumes:
      - db_data:/var/lib/postgresql/data

  fusionauth:
    image: fusionauth/fusionauth-app:latest
    depends_on:
      - db
      - kafka
      - zookeeper
    environment:
      DATABASE_URL: jdbc:postgresql://db:5432/fusionauth
      DATABASE_ROOT_USER: ${POSTGRES_USER}
      DATABASE_ROOT_PASSWORD: ${POSTGRES_PASSWORD}
      DATABASE_USER: ${DATABASE_USER}
      DATABASE_PASSWORD: ${DATABASE_PASSWORD}
#      FUSIONAUTH_MEMORY: ${FUSIONAUTH_MEMORY}
      FUSIONAUTH_SEARCH_ENGINE_TYPE: database
      FUSIONAUTH_URL: http://fusionauth:9011
      FUSIONAUTH_RUNTIME_MODE: development
    networks:
     - mynetwork
    restart: unless-stopped
    ports:
      - 9011:9011
    volumes:
      - fa_config:/usr/local/fusionauth/config

networks:
  mynetwork:
    driver: bridge

volumes:
  db_data:
  fa_config:
```
If I edit /etc/hosts and add kafka as an alias to 127.0.0.1, I can list the topics using the kafka command line.

I see this in the logs (just what @joydeb28 sees) whenever I try to send a test message in the logs:

```
fusionauth_1  | org.apache.kafka.common.config.ConfigException: Missing required configuration "bootstrap.servers" which has no default value.
fusionauth_1  |         at org.apache.kafka.common.config.ConfigDef.parseValue(ConfigDef.java:463)
fusionauth_1  |         at org.apache.kafka.common.config.ConfigDef.parse(ConfigDef.java:453)
fusionauth_1  |         at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:62)
fusionauth_1  |         at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:75)
fusionauth_1  |         at org.apache.kafka.clients.producer.ProducerConfig.<init>(ProducerConfig.java:359)
fusionauth_1  |         at org.apache.kafka.clients.producer.KafkaProducer.<init>(KafkaProducer.java:300)
fusionauth_1  |         at io.fusionauth.api.service.messaging.KafkaProducerBuilder.build(KafkaProducerBuilder.java:30)
fusionauth_1  |         at io.fusionauth.api.service.messaging.KafkaProducerProvider.onCacheLoad(KafkaProducerProvider.java:43)
fusionauth_1  |         at com.inversoft.cache.BaseCacheLoader.load(BaseCacheLoader.java:24)
fusionauth_1  |         at io.fusionauth.api.service.cache.IntegrationConfigurationCacheLoader.run(IntegrationConfigurationCacheLoader.java:36)
fusionauth_1  |         at com.inversoft.scheduler.InjectableRunnable.run(InjectableRunnable.java:37)
fusionauth_1  |         at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
fusionauth_1  |         at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
fusionauth_1  |         at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
fusionauth_1  |         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
fusionauth_1  |         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
fusionauth_1  |         at java.base/java.lang.Thread.run(Thread.java:832)
```

And I see this message in the admin screen:
```
org.apache.kafka.common.config.ConfigException: Missing required configuration "bootstrap.servers" which has no default value.
  - Missing required configuration "bootstrap.servers" which has no default value.
```

If I get the integration configuration from the API (via curl), I see:

```
{"integrations":{"cleanspeak":{"enabled":false,"usernameModeration":{"enabled":false}},"kafka":{"defaultTopic":"test","enabled":true,"producer":{"bootstrap.servers":"kafka:9092","max.block.ms":"10000","request.timeout.ms":"2000"}},"twilio":{"enabled":false,"url":"https://api.twilio.com"}}}
```

@joydeb28 I don't see anything you are doing wrong, I think this is a bug.

**Comment by robotdan** [comment]
Can you show your log right after you save your Kafka configuration? It seems to me the cache reload isn't working. 

To test this theory, try these steps and report back on each test. 

1. Edit your Kafka configuration, save, wait 2 minutes, enter the config and use the Test button
2. Edit your Kafka configuration w/out saving, use the Test button before saving.

**Comment by joydeb28** [comment]
Sorry for the delayed response.
LOGS:
1. 
```
fusionauth_1  | Jun 12, 2020 4:05:38.637 PM INFO  io.fusionauth.api.service.messaging.KafkaProducerProvider - Rebuild Kafka Producer
fusionauth_1  | Jun 12, 2020 4:05:38.640 PM ERROR io.fusionauth.app.primeframework.error.ExceptionExceptionHandler - An unhandled exception was thrown
fusionauth_1  | org.apache.kafka.common.config.ConfigException: Missing required configuration "bootstrap.servers" which has no default value.
fusionauth_1  | 	at org.apache.kafka.common.config.ConfigDef.parseValue(ConfigDef.java:463)
fusionauth_1  | 	at org.apache.kafka.common.config.ConfigDef.parse(ConfigDef.java:453)
fusionauth_1  | 	at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:62)
fusionauth_1  | 	at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:75)
fusionauth_1  | 	at org.apache.kafka.clients.producer.ProducerConfig.<init>(ProducerConfig.java:359)
fusionauth_1  | 	at org.apache.kafka.clients.producer.KafkaProducer.<init>(KafkaProducer.java:300)
fusionauth_1  | 	at io.fusionauth.api.service.messaging.KafkaProducerBuilder.build(KafkaProducerBuilder.java:30)
fusionauth_1  | 	at io.fusionauth.api.service.messaging.KafkaProducerProvider.onCacheLoad(KafkaProducerProvider.java:43)
fusionauth_1  | 	at com.inversoft.cache.BaseCacheLoader.load(BaseCacheLoader.java:24)
fusionauth_1  | 	at io.fusionauth.app.action.api.cache.ReloadAction.post(ReloadAction.java:40)
fusionauth_1  | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
fusionauth_1  | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
fusionauth_1  | 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
fusionauth_1  | 	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
fusionauth_1  | 	at org.primeframework.mvc.util.ReflectionUtils.invoke(ReflectionUtils.java:414)
fusionauth_1  | 	at org.primeframework.mvc.action.DefaultActionInvocationWorkflow.execute(DefaultActionInvocationWorkflow.java:79)
fusionauth_1  | 	at org.primeframework.mvc.action.DefaultActionInvocationWorkflow.perform(DefaultActionInvocationWorkflow.java:62)
fusionauth_1  | 	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
fusionauth_1  | 	at org.primeframework.mvc.validation.DefaultValidationWorkflow.perform(DefaultValidationWorkflow.java:47)
fusionauth_1  | 	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
fusionauth_1  | 	at org.primeframework.mvc.security.DefaultSecurityWorkflow.perform(DefaultSecurityWorkflow.java:81)
fusionauth_1  | 	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
fusionauth_1  | 	at org.primeframework.mvc.parameter.DefaultPostParameterWorkflow.perform(DefaultPostParameterWorkflow.java:50)
fusionauth_1  | 	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
fusionauth_1  | 	at org.primeframework.mvc.content.DefaultContentWorkflow.perform(DefaultContentWorkflow.java:52)
fusionauth_1  | 	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
fusionauth_1  | 	at org.primeframework.mvc.parameter.DefaultParameterWorkflow.perform(DefaultParameterWorkflow.java:57)
fusionauth_1  | 	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
fusionauth_1  | 	at org.primeframework.mvc.parameter.DefaultURIParameterWorkflow.perform(DefaultURIParameterWorkflow.java:102)
fusionauth_1  | 	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
fusionauth_1  | 	at org.primeframework.mvc.scope.DefaultScopeRetrievalWorkflow.perform(DefaultScopeRetrievalWorkflow.java:58)
fusionauth_1  | 	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
fusionauth_1  | 	at org.primeframework.mvc.message.DefaultMessageWorkflow.perform(DefaultMessageWorkflow.java:45)
fusionauth_1  | 	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
fusionauth_1  | 	at io.fusionauth.app.primeframework.FrontEndTenantWorkflow.perform(FrontEndTenantWorkflow.java:71)
fusionauth_1  | 	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
fusionauth_1  | 	at org.primeframework.mvc.action.DefaultActionMappingWorkflow.perform(DefaultActionMappingWorkflow.java:126)
fusionauth_1  | 	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
fusionauth_1  | 	at org.primeframework.mvc.workflow.StaticResourceWorkflow.perform(StaticResourceWorkflow.java:97)
fusionauth_1  | 	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
fusionauth_1  | 	at org.primeframework.mvc.parameter.RequestBodyWorkflow.perform(RequestBodyWorkflow.java:89)
fusionauth_1  | 	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
fusionauth_1  | 	at org.primeframework.mvc.security.DefaultSavedRequestWorkflow.perform(DefaultSavedRequestWorkflow.java:57)
fusionauth_1  | 	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
fusionauth_1  | 	at io.fusionauth.app.primeframework.CORSFilter.handleNonCORS(CORSFilter.java:291)
fusionauth_1  | 	at io.fusionauth.app.primeframework.CORSFilter.doFilter(CORSFilter.java:252)
fusionauth_1  | 	at io.fusionauth.app.primeframework.CORSRequestWorkflow.perform(CORSRequestWorkflow.java:50)
fusionauth_1  | 	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
fusionauth_1  | 	at io.fusionauth.app.primeframework.FusionAuthMVCWorkflow.perform(FusionAuthMVCWorkflow.java:88)
fusionauth_1  | 	at org.primeframework.mvc.workflow.DefaultWorkflowChain.continueWorkflow(DefaultWorkflowChain.java:44)
fusionauth_1  | 	at org.primeframework.mvc.servlet.FilterWorkflowChain.continueWorkflow(FilterWorkflowChain.java:50)
fusionauth_1  | 	at org.primeframework.mvc.servlet.PrimeFilter.doFilter(PrimeFilter.java:82)
fusionauth_1  | 	at com.inversoft.maintenance.servlet.MaintenanceModePrimeFilter.doFilter(MaintenanceModePrimeFilter.java:59)
fusionauth_1  | 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
fusionauth_1  | 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
fusionauth_1  | 	at com.inversoft.servlet.UTF8Filter.doFilter(UTF8Filter.java:27)
fusionauth_1  | 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
fusionauth_1  | 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
fusionauth_1  | 	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:199)
fusionauth_1  | 	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
fusionauth_1  | 	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:543)
fusionauth_1  | 	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
fusionauth_1  | 	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81)
fusionauth_1  | 	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
fusionauth_1  | 	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
fusionauth_1  | 	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:615)
fusionauth_1  | 	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
fusionauth_1  | 	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:818)
fusionauth_1  | 	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1623)
fusionauth_1  | 	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
fusionauth_1  | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
fusionauth_1  | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
fusionauth_1  | 	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
fusionauth_1  | 	at java.base/java.lang.Thread.run(Thread.java:832)
fusionauth_1  | Jun 12, 2020 4:05:38.647 PM ERROR io.fusionauth.api.service.cache.DistributedCacheNotifier - Failed to request a cache reload for [http://fusionauth:9011]. Status Code [500]
fusionauth_1  | Jun 12, 2020 4:05:55.697 PM INFO  io.fusionauth.api.service.messaging.KafkaProducerProvider - Rebuild Kafka Producer
fusionauth_1  | Jun 12, 2020 4:05:55.702 PM ERROR com.inversoft.scheduler.LogAndRetainFailureHandler - The scheduled service [class io.fusionauth.api.service.cache.IntegrationConfigurationCacheLoader] failed but will be re-run.
fusionauth_1  | org.apache.kafka.common.config.ConfigException: Missing required configuration "bootstrap.servers" which has no default value.
fusionauth_1  | 	at org.apache.kafka.common.config.ConfigDef.parseValue(ConfigDef.java:463)
fusionauth_1  | 	at org.apache.kafka.common.config.ConfigDef.parse(ConfigDef.java:453)
fusionauth_1  | 	at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:62)
fusionauth_1  | 	at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:75)
fusionauth_1  | 	at org.apache.kafka.clients.producer.ProducerConfig.<init>(ProducerConfig.java:359)
fusionauth_1  | 	at org.apache.kafka.clients.producer.KafkaProducer.<init>(KafkaProducer.java:300)
fusionauth_1  | 	at io.fusionauth.api.service.messaging.KafkaProducerBuilder.build(KafkaProducerBuilder.java:30)
fusionauth_1  | 	at io.fusionauth.api.service.messaging.KafkaProducerProvider.onCacheLoad(KafkaProducerProvider.java:43)
fusionauth_1  | 	at com.inversoft.cache.BaseCacheLoader.load(BaseCacheLoader.java:24)
fusionauth_1  | 	at io.fusionauth.api.service.cache.IntegrationConfigurationCacheLoader.run(IntegrationConfigurationCacheLoader.java:36)
fusionauth_1  | 	at com.inversoft.scheduler.InjectableRunnable.run(InjectableRunnable.java:37)
fusionauth_1  | 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
fusionauth_1  | 	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
fusionauth_1  | 	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
fusionauth_1  | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
fusionauth_1  | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
fusionauth_1  | 	at java.base/java.lang.Thread.run(Thread.java:832)
```


2. 
```
fusionauth_1  | Jun 12, 2020 4:07:41.697 PM INFO  io.fusionauth.api.service.messaging.KafkaProducerProvider - Rebuild Kafka Producer
fusionauth_1  | Jun 12, 2020 4:07:41.698 PM ERROR com.inversoft.scheduler.LogAndRetainFailureHandler - The scheduled service [class io.fusionauth.api.service.cache.IntegrationConfigurationCacheLoader] failed but will be re-run.
fusionauth_1  | org.apache.kafka.common.config.ConfigException: Missing required configuration "bootstrap.servers" which has no default value.
fusionauth_1  | 	at org.apache.kafka.common.config.ConfigDef.parseValue(ConfigDef.java:463)
fusionauth_1  | 	at org.apache.kafka.common.config.ConfigDef.parse(ConfigDef.java:453)
fusionauth_1  | 	at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:62)
fusionauth_1  | 	at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:75)
fusionauth_1  | 	at org.apache.kafka.clients.producer.ProducerConfig.<init>(ProducerConfig.java:359)
fusionauth_1  | 	at org.apache.kafka.clients.producer.KafkaProducer.<init>(KafkaProducer.java:300)
fusionauth_1  | 	at io.fusionauth.api.service.messaging.KafkaProducerBuilder.build(KafkaProducerBuilder.java:30)
fusionauth_1  | 	at io.fusionauth.api.service.messaging.KafkaProducerProvider.onCacheLoad(KafkaProducerProvider.java:43)
fusionauth_1  | 	at com.inversoft.cache.BaseCacheLoader.load(BaseCacheLoader.java:24)
fusionauth_1  | 	at io.fusionauth.api.service.cache.IntegrationConfigurationCacheLoader.run(IntegrationConfigurationCacheLoader.java:36)
fusionauth_1  | 	at com.inversoft.scheduler.InjectableRunnable.run(InjectableRunnable.java:37)
fusionauth_1  | 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
fusionauth_1  | 	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
fusionauth_1  | 	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
fusionauth_1  | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
fusionauth_1  | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
fusionauth_1  | 	at java.base/java.lang.Thread.run(Thread.java:832)
```

**Comment by joydeb28** [comment]
Hi @robotdan and @mooreds 

Any update on this??

**Comment by robotdan** [comment]
I'll take a look at this shortly.

**Comment by robotdan** [comment]
Fixed in version 1.17.5. Let me know how it goes! Thanks for reporting.
